src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGithubForDevelopment.java

```diff
@@ -39,31 +56,100 @@ protected Value fetchValueFor(GitHubProject project) throws IOException {
   }
 
   /**
-   * Checks if the project uses GitHub as a main development platform.
+   * Checks if a project uses GitHub as a main development platform.
    *
-   * @param project of type {@link GitHubProject}.
-   * @return {@link BooleanValue} value of the feature.
+   * @param project The project to be checked.
+   * @return {@link BooleanValue} of the feature.
    */
   private Value<Boolean> usesGithubForDevelopment(GitHubProject project) {
-    // TODO: This is just a basic check, more constraints have to be considered in the future with
-    //       more clarity. Please refer https://github.com/SAP/fosstars-rating-core/issues/122
     try {
-      return USES_GITHUB_FOR_DEVELOPMENT.value(!isMirror(project));
+      GHRepository repository = gitHubDataFetcher().repositoryFor(project, github);
+
+      Date threeMonthsAgo = Date.from(Instant.now().minus(3, ChronoUnit.MONTHS));
+      List<GHCommit> commits = gitHubDataFetcher().commitsAfter(threeMonthsAgo, project, github);
+
+      return USES_GITHUB_FOR_DEVELOPMENT.value(
+          usesGitHubForDevelopment(repository, commits, CONFIDENCE_THRESHOLD));
     } catch (IOException e) {
       logger.warn("Couldn't fetch data, something went wrong!", e);
       return USES_GITHUB_FOR_DEVELOPMENT.unknown();
     }
   }
 
   /**
-   * Checks if the input GitHub project is a mirror or an original project.
+   * The method checks if it looks like that a project uses GitHub as a main development platform.
+   * The method runs a number of checks for the project. If most of the checks pass,
+   * then the method concludes that the projects uses GitHub for development.
    * 
-   * @param project of type {@link GitHubProject}.
-   * @return true if the project is a mirror project. Otherwise, false.
-   * @throws IOException may throw an exception during REST call to GitHub API.
+   * @param repository The project's repository.
+   * @param commits The project's commit history.
+   * @return True if it looks like that the project uses GitHub, false otherwise.
+   *
+   */
+  boolean usesGitHubForDevelopment(
+      GHRepository repository, List<GHCommit> commits, double threshold) {
+
+    List<Predicate<GHRepository>> checks = Arrays.asList(
```

Comment:
This arraylist can be static as well.

Reply:
Not really at the moment. The lambdas in the array use the `commits` parameter but the `Predicate` interface takes only one parameter and it is `GHRepository`. To overcome this, I need to introduce a new interface for a check. The new interface should take both repository and list of commits as a parameter. At first, I thought using the `Predicate` interface is better (and shorter). I'll introduce a new interface for checks.